web/src/main/java/org/wso2/testgrid/web/api/APIUtil.java

```diff
@@ -157,18 +163,20 @@ static TestPlan getTestPlanBean(org.wso2.testgrid.common.TestPlan testPlan, bool
     /**
      * Util method to convert {@link org.wso2.testgrid.common.TestScenario} instance to a {@link TestScenario} instance.
      *
-     * @param testScenario {@link org.wso2.testgrid.common.TestScenario} instance to be converted
+     * @param testScenario     {@link org.wso2.testgrid.common.TestScenario} instance to be converted
      * @param requireTestCases boolean flag to indicate whether to include test-case info
      * @return {@link TestScenario} instance with necessary information
      */
     static TestScenario getTestScenarioBean(org.wso2.testgrid.common.TestScenario testScenario,
                                             boolean requireTestCases) {
         TestScenario testScenario1 = new TestScenario();
-        testScenario1.setId(testScenario.getId());
-        testScenario1.setName(testScenario.getName());
-        testScenario1.setStatus(testScenario.getStatus().toString());
-        if (requireTestCases) {
-            testScenario1.setTestCases(getTestCaseBeans(testScenario.getTestCases()));
+        if (testScenario != null) {
+            testScenario1.setId(testScenario.getId());
```

Comment:
Shall we change the variable names `testScenario` and `testScenario1` to something more meaningful since it is hard to distinguish which one is which at the moment